### ISSUE TITLE
NullPointerException: Attempt to invoke virtual method 'void android.graphics.Bitmap.recycle()' on a null object reference

### ISSUE THREAD

**Issue by ugurcany** [wasabeef/Blurry#50]
I'm getting the following err:
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.graphics.Bitmap.recycle()' on a null object reference
       at jp.wasabeef.blurry.internal.Blur.of(SourceFile:42)
       at jp.wasabeef.blurry.Blurry$Composer.onto(SourceFile:115)
       ...................
```

...while calling the following:
```
Blurry.with(activity)
                                .radius(40)
                                .sampling(2)
                                .color(ResourcesUtil.getColor(R.color.dirtyBlack25))
                                .onto(blurredView);
```

Do you have any idea?

**Comment by HugoMatilla** [comment]
Same here.

**Comment by micutad** [comment]
+1

**Comment by zxx714** [comment]
@ugurcany When did you call Blurry.with? Is it too early? I mean before the view is attached to window? 

However I think Blurry shouldn't make app crash or give some meaningful exception under this case at least.

**Comment by ugurcany** [comment]
In my case, the creation of blur view is done inside `onShowListener()` of a dialog that I create. And the dialog creation is triggered by a user input.

**Comment by zxx714** [comment]
@ugurcany Did you use Blurry to blur the background of your dialog?

What's the blurredView you passed in your code?

**Comment by ugurcany** [comment]
@zxx714 Yes, we can say that. I'm passing the rootview of my Activity as the view to be blurred. Just want to mention again that I rarely encounter this issue.

**Comment by KISINGA** [comment]
For anybody facing this problem, this is probably related to running the code too soon after the view is loaded
try this attach a runnable
 imageView.post(new Runnable() {
                @Override
                public void run() {
                    Blurry.with(Payment.this)
                            .radius(25)
                            .sampling(2)
                            .animate(500)
                            .onto(payments);
                }
            });

**Comment by AlohaQwQ** [comment]
I found the following code from the source code.
   public static Bitmap of(View view, BlurFactor factor) {
    view.setDrawingCacheEnabled(true);
    view.destroyDrawingCache();
    view.setDrawingCacheQuality(View.DRAWING_CACHE_QUALITY_LOW);
    Bitmap cache = view.getDrawingCache();
    Bitmap bitmap = of(view.getContext(), cache, factor);
    cache.recycle();
    return bitmap;
  }
I don't know why call destroyDrawingCache() then call view.getDrawingCache(), so that may the cache is null,and then call cache.recycle(). This problem is caused by the difference version?

**Comment by ameersabith** [comment]
I have a problem while calling `notifyDataSetChanged();`.

Error: 
>Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.example.sabith.restorentbiller.ItemsViewAdapter.notifyDataSetChanged()' on a null object reference

**Comment by MensObscura** [comment]
Hello, i tried to use it on a FrameLayout, Containing a GLSurfaceView Which is displaying the Camera Stream.
First i got the '"NullPointerException: Attempt to invoke virtual method 'void android.graphics.Bitmap.recycle()'"

And then when I try using the pos(...)t method, it just does have no effect =) I 'assume this lib can't work with non-fixed background =)

**Comment by snowf07** [comment]
My solution，

    handler.postDelayed(new Runnable() {

            @Override
            public void run() {
                Blurry.with(mContext).radius(25).capture(mImgAudio).into(mImgBlurryfloat);

            }
        },1000);

**Comment by jasonrobinson-gpsw** [comment]
Kotlin approach:

```
private fun Blurry.Composer.postOnto(view: ViewGroup) {
    view.post { onto(view) }
}
```

**Comment by Vovchikus-RedTail** [comment]
.async() do the trick
So in the result it looks like this: Blurry.with(context).{other calls here}.async().onto(view)

**Comment by sosmallsotiny** [comment]
I faced same problem and tried all of above solutions but none of them works :(
Also since "view.getDrawingCache()"has become decrypt since android API30, 
it seems necessary to delete all of the cache process...

**Comment by Oshuremy** [comment]
I'm facing this problem event with kotlin approach: 

```
private fun Blurry.Composer.postOnto(view: ViewGroup) {
    view.post { onto(view) }
}
```
